src/cc/frontends/clang/b_frontend_action.cc

```diff
@@ -523,6 +524,54 @@ bool BTypeVisitor::VisitCallExpr(CallExpr *Call) {
   return true;
 }
 
+bool BTypeVisitor::checkFormatSpecifiers(string fmt, SourceLocation loc) {
+  unsigned nb_specifiers = 0, i, j;
+  bool has_s = false;
+  for (i = 0; i < fmt.length(); i++) {
+    if (!isascii(fmt[i]) || (!isprint(fmt[i]) && !isspace(fmt[i]))) {
+      warning(loc.getLocWithOffset(i), "unrecognized character");
+      return false;
+    }
+    if (fmt[i] != '%')
+      continue;
+    if (nb_specifiers >= 3) {
```

Comment:
Seems like printing literal `"%%"` isn't supported here or in the kernel. Is that the right interpretation?